### PR TITLE
Fixes #8429: rudder-dev merge --test can fail

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/8429

### CODE REVIEW

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
This is a test

File: scripts/rudder-dev/rudder-dev-src [hunk 1]
```diff
@@ -846,6 +848,8 @@ def insert_line(filename, regex, text):
 # fetch a branch from a PR
 def fetch_branch_from_pr(pr):
   branch_name = pr.remote_branch()
+  # PR can be rebased, first destroy existing branch
+  shell("git branch -D " + branch_name + " 2>/dev/null || true", "Cleaning existing fetched branch")
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/rudder-dev-src]: It should not be branch_name that should be deleted but a branch created for the fetch (branch_name_remote ? )

**Comment by peckpeck** [comment]
Commit modified

File: scripts/rudder-dev/rudder-dev-src [hunk 2]
```diff
@@ -846,7 +848,8 @@ def insert_line(filename, regex, text):
 # fetch a branch from a PR
 def fetch_branch_from_pr(pr):
   branch_name = pr.remote_branch()
-  shell("git fetch --update-head-ok " + pr.repo() + " " + branch_name + ":" + branch_name, "Fetching branch from remote " + pr.repo())
+  # PR can be rebased, first destroy existing branch
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/rudder-dev-src]: The commentis noit accurate, we don't destroy branch but force update on a dedicated branch

**Comment by peckpeck** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Can be merged :)

**Pull request merged**